### ISSUE TITLE
Disposing ValueBinding does not remove listeners placed on its model and target

### ISSUE THREAD

**Issue by habbit-baggins** [eclipse-platform/eclipse.platform.ui#1114]
The code in ValueBinding that nullifies `target` and `model` conflicts with the later call to `super.dispose()`, preventing Binding (the superclass) from removing the "on dispose" listeners that it might have placed on those objects.

https://github.com/eclipse-platform/eclipse.platform.ui/blob/8982941a20a264e7c06a9585006f35af1709f7b4/bundles/org.eclipse.core.databinding/src/org/eclipse/core/databinding/ValueBinding.java#L260-L262

https://github.com/eclipse-platform/eclipse.platform.ui/blob/8982941a20a264e7c06a9585006f35af1709f7b4/bundles/org.eclipse.core.databinding/src/org/eclipse/core/databinding/Binding.java#L137-L147

The effect does _not_ seem to be deliberate, since:
* The change to ValueBinding was earlier, in commit e52e28d41bd4e6ac2b7d02d50fba457e69961880, while the dispose listener was not introduced to until a bit later, in commit 7978ebc16da1efae2a5bf089c64e76675b8b5b2e.
* ListBinding and SetBinding have a very similar `dispose` implementation, but they do _not_ nullify `target` and `model` themselves.

**Comment by BeckerWdf** [comment]
So would the proposal be to swapp the "nullifiying" so that it happens after the super.disose() call?

**Comment by habbit-baggins** [comment]
I was going to say that it would not be required since the parent class does nullify them. However, I just noticed that both Binding and ValueBinding have _separate_ fields `target` and `model`. Thus, the code I highlighted in ValueBinding is _not_ actually nullifying the variables in Binding.

Welp, that means that my own issue (that I though arose from this) is actually coming from elsewhere.

I am withdrawing/closing the issue as-is, but perhaps the field names in ValueBinding should be changed to avoid the confusion in the future. For example, they could be named targetVal/modelVal, since the only difference with the same fields in the parent class is the type (IObservable vs IObservableValue).

**Comment by sratz** [comment]
Both `ValueBinding` and `Binding` keep there **_own_** fields of `target` and `model`,

Nullifying in both sub and super class is therefore correct, as nullifying the fields of `ValueBinding` has no effect on the fields of the super class `Binding`, hence, `super.dispose()` is completely unaffected by this.

What exactly is the problem you are seeing?

**Comment by habbit-baggins** [comment]
The problem I am seeing is that some SWT Text (with associated ControlDecorationSupport) is being called for update after the binding is disposed, so I was investigating in the binding-related classes. When I saw these variables being nullified in ValueBinding _but not List/SetBinding_, I thought I was on to something, but clearly I failed to notice that Binding and ValueBinding had different fields with the same names.

Sorry for wasting your time, folks, I will keep investigating on my side.
